k4Reco/Overlay/components/OverlayTiming.cpp

```diff
@@ -378,4 +381,24 @@ retType OverlayTiming::operator()(const edm4hep::EventHeaderCollection&
                          std::move(ocaloHitContribs));
 }
 
+StatusCode OverlayTiming::finalize() {
+  if (m_copyCellIDMetadata) {
+    for (auto& [input, output] :
+         {std::make_pair(inputLocations("SimTrackerHits"), outputLocations("OutputSimTrackerHits")),
+          std::make_pair(inputLocations("SimCalorimeterHits"), outputLocations("OutputSimCalorimeterHits"))}) {
+      for (size_t i = 0; i < input.size(); ++i) {
+        auto name  = input[i];
+        auto value = k4FWCore::getParameter<std::string>(name + "__" + edm4hep::labels::CellIDEncoding, this);
```

Comment:
```suggestion
        auto value = k4FWCore::getParameter<std::string>(podio::collMetaDataParamName(name, edm4hep::labels::CellIDEncoding), this);
```

Gives as an easier time figuring out where we deal with collection parameters. Alternatively, we could also make an overload for `k4FWCore::getParameter` that either takes an additional collection name, or a collection pointer and we hide all of the details in there.

Reply:
Done